data/colors/vars/component_dark.ts

```diff
@@ -167,5 +167,24 @@ export default {
       activeBg: alpha(get('scale.red.4'), 0.24),
       hoverText: get('scale.red.3')
     }
+  },
+
+  switchTrack: {
+    bg: get('scale.black'),
+    border: get('scale.gray.3'),
+
+    checked: {
+      bg: alpha(get('scale.blue.5'), 0.35),
+      hoverBg: alpha(get('scale.blue.5'), 0.5),
+      activeBg: alpha(get('scale.blue.5'), 0.65),
+      border: get('scale.blue.3'),
+    }
+  },
+
+  switchKnob: {
```

Comment:
Curious: How do we decide the nesting structure in primitives, are there any rules/ADRs around it?

switch: {
  track: {},
  knob: {}
}

vs 

switchTrack: {},
switchKnob: {}

Reply:
I'm not aware of any rules around this. I considered nesting `switch.track` and `switch.knob` too, but decided to keep it flatter.

I don't feel strongly either way.